File: iOS/ssabree/ssabree/Core/Repository/CommentRepositoryImpl.swift
import Foundation

final class CommentRepositoryImpl: CommentRepository {
    private let commentService: CommentService

    init(commentService: CommentService) {
        self.commentService = commentService
    }

    func like(commentId: Int) async -> Result<CommentLikeModel, Error> {
        do {
            let response = try await commentService.likeComment(commentId: commentId)
            return .success(response.toRepositoryModel())
        } catch {
            return .failure(error)
        }
    }

    func unlike(commentId: Int) async -> Result<CommentLikeModel, Error> {
        do {
            let response = try await commentService.unlikeComment(commentId: commentId)
            return .success(response.toRepositoryModel())
        } catch {
            return .failure(error)
        }
    }

    func updateComment(commentId: Int, content: String) async -> Result<Void, Error> {
        do {
            try await commentService.updateComment(commentId: commentId, content: content)
            return .success(())
        } catch {
            return .failure(error)
        }
    }

    func deleteComment(commentId: Int) async -> Result<Void, Error> {
        do {
            try await commentService.deleteComment(commentId: commentId)
            return .success(())
        } catch {
            return .failure(error)
        }
    }
}

extension CommentLikeResponse {
    func toRepositoryModel() -> CommentLikeModel {
        CommentLikeModel(liked: liked, likeCount: likeCount)
    }
}

File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/repository/CommentRepositoryImpl.kt
package com.ssafy.ssabree.core.repository

import com.ssafy.ssabree.core.datasource.remote.CommentService
import com.ssafy.ssabree.core.datasource.remote.model.CommentUpdateRequest
import com.ssafy.ssabree.core.repository.model.CommentLikeModel
import com.ssafy.ssabree.core.repository.model.toModel
import com.ssafy.ssabree.core.utils.RetrofitClient

class CommentRepositoryImpl : CommentRepository {

    private val commentService = RetrofitClient.instance.create(CommentService::class.java)

    override suspend fun updateComment(commentId: Long, content: String): Result<Unit> {
        return runCatching {
            commentService.updateComment(commentId, CommentUpdateRequest(content))
        }
    }

    override suspend fun deleteComment(commentId: Long): Result<Unit> {
        return runCatching {
            commentService.deleteComment(commentId)
        }
    }

    override suspend fun like(commentId: Long): Result<CommentLikeModel> {
        return runCatching {
            commentService.like(commentId).toModel()
        }
    }

    override suspend fun unlike(commentId: Long): Result<CommentLikeModel> {
        return runCatching {
            commentService.unlike(commentId).toModel()
        }
    }
}
